pkgs/by-name/mn/mnamer/package.nix

```diff
@@ -1,31 +1,52 @@
-{ python3Packages, fetchFromGitHub, lib }:
+{
+  python3Packages,
+  fetchFromGitHub,
+  fetchpatch,
+  lib,
+}:
 
 python3Packages.buildPythonApplication rec {
   pname = "mnamer";
-  version = "2.5.4";
+  version = "2.5.5";
+  format = "pyproject";
 
   src = fetchFromGitHub {
     owner = "jkwill87";
     repo = "mnamer";
     rev = version;
-    sha256 = "sha256-fONQq/RboWHFuEFU7HP1ThUpSjOIlkg54c2WlMUKwuk=";
+    sha256 = "sha256-qQu5V1GOsbrR00HOrot6TTAkc3KRasBPDEU7ZojUBio=";
   };
 
+  build-system = with python3Packages; [
+    setuptools
+    setuptools-scm
+  ];
+
+  nativeBuildInputs = with python3Packages; [ pythonRelaxDepsHook ];
+
   propagatedBuildInputs = with python3Packages; [
+    appdirs
     babelfish
+    guessit
     requests
-    appdirs
-    teletype
     requests-cache
-    guessit
+    teletype
   ];
 
-  patches = [
-    # requires specific old versions of dependencies which have been updated in nixpkgs
-    ./remove_requirements.patch
+  pythonRelaxDeps = [
+    "guessit"
+    "requests-cache"
+    "setuptools-scm"
+    "typing-extensions"
+  ];
 
-    # author reads a private property that changed between versions
-    ./update_hack.patch
+  patches = [
+    # https://github.com/jkwill87/mnamer/pull/291
+    (fetchpatch {
+      name = "fix-cached-session-object-error";
+      url = "https://patch-diff.githubusercontent.com/raw/jkwill87/mnamer/pull/291.patch";
+      hash = "sha256-b0mD9lLE0cUy3kk6Y04MjJfBXSj8HDApmo4xoBINAwA=";
+    })
```

Comment:
I am skeptical of this patch, but I don't know enough about the project to judge whether this is a valid fix or not. Is that initial session state needed?

Reply:
I haven't looked into it deeply, but I know from actual usage of the program that it doesn't work at all as-is, it works as expected with the patch.